gen/src/Gen/Types/Retry.hs

```diff
@@ -59,63 +57,69 @@ data Delay = Delay
   }
   deriving (Eq, Show, Generic)
 
-makeClassy ''Delay
+$(Lens.makeClassy ''Delay)
 
 instance FromJSON Delay where
-  parseJSON = withObject "delay" $ \o ->
-    Delay <$> upperHead <$> o .: "type"
-      <*> (o .: "base" >>= base)
-      <*> o .: "growth_factor"
+  parseJSON =
+    Aeson.withObject "Delay" $ \o ->
+      Delay <$> upperHead <$> o .: "type"
+        <*> (o .: "base" >>= base)
+        <*> o .: "growth_factor"
     where
       base = \case
-        String "rand" -> pure 0.05
-        o -> parseJSON o
+        Aeson.String "rand" -> pure 0.05
+        o -> Aeson.parseJSON o
 
 data Retry = Retry'
   { _retryAttempts :: Integer,
     _retryDelay :: Delay,
-    _retryPolicies :: Map Text Policy
+    _retryPolicies :: HashMap Text Policy
   }
   deriving (Eq, Show)
 
-makeLenses ''Retry
+$(Lens.makeLenses ''Retry)
 
 instance HasDelay Retry where
   delay = retryDelay
 
 instance FromJSON Retry where
-  parseJSON = withObject "default_retry" $ \o ->
-    Retry' <$> o .: "max_attempts"
-      <*> o .: "delay"
-      <*> pure mempty
-
-instance FromJSON (Retry -> Retry) where
-  parseJSON = withObject "retry" $ \o -> do
-    m <- o .:? "max_attempts"
-    d <- o .:? "delay"
-    -- FIXME: Currently simply ignoring non '__default__' keys.
-    p <-
-      (o .: defKey <|> pure mempty)
-        >>= (\o' -> o' .: "policies" <|> pure mempty)
-    return $ \r ->
+  parseJSON =
+    Aeson.withObject "default_retry" $ \o ->
       Retry'
-        (fromMaybe (r ^. retryAttempts) m)
-        (fromMaybe (r ^. retryDelay) d)
-        (r ^. retryPolicies <> p)
+        <$> o .: "max_attempts"
+        <*> o .: "delay"
+        <*> pure mempty
 
-parseRetry :: Text -> Object -> Parser Retry
+instance FromJSON (Retry -> Retry) where
```

Comment:
I generally find `FromJSON` instances on functions very confusing, but sometimes there's no better way.

Reply:
If this is the most confusing thing you've found in the generator - I applaud you, Sir.

Reply:
TBQH I've found the generator reasonably comprehensible - when I've needed to tweak something the lever I needed seemed to be pretty easy to find.

But maybe that's because I used to understand the autotools and it's permanently damaged my baseline.